library/std/src/thread/builder.rs

```diff
@@ -212,10 +220,18 @@ impl Builder {
     /// [`io::Result`] to capture any failure to create the thread at
     /// the OS level.
     ///
+    /// Like [`spawn`], this method will still call the main thread functions
+    /// added by [`add_spawn_hook`] (unless [`Builder::no_hooks`] was called),
+    /// but won't execute the returned functions if thread creation fails at the
+    /// OS level.
+    ///
     /// # Panics
     ///
     /// Panics if a thread name was set and it contained null bytes.
     ///
+    /// Unlike the [`spawn`] free function, if it panics for this reason, the
```

Comment:
> Unlike the `spawn` free function

`thread::spawn` calls `Builder::spawn` internally, so this is misleading – it's just that `thread::spawn` doesn't set a thread name and thus will never panic for this reason.